lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -0,0 +1,590 @@
+      const error = new Error(`Could not broadcast transaction ${transaction.toString()}`);
+      console.error(error);
+      throw error;
+    }
+    console.info(`Successfully submitted transaction ${transaction.id}`);
+  }
+
+  /**
+   * Gets all unspent coins of a given address
+   * @param address Bitcoin address to get coins for
+   */
+  private async getUnspentCoins (address: Address): Promise<Transaction.UnspentOutput[]> {
+    const addressToSearch = address.toString();
+    console.info(`Getting unspent coins for ${addressToSearch}`);
+    const requestPath = `/coin/address/${addressToSearch}`;
+
+    const fullPath = new URL(requestPath, this.bitcoinExtensionUri);
+    const response = await this.fetchWithRetry(fullPath.toString());
+
+    const responseData = await ReadableStreamUtils.readAll(response.body);
+    if (response.status !== httpStatus.OK) {
+      const error = new Error(`Fetch failed [${response.status}]: ${responseData}`);
+      console.error(error);
+      throw error;
+    }
+
+    const responseJson = JSON.parse(responseData) as Array<any>;
+    const unspentTransactions = responseJson.map((coin) => {
+      return new Transaction.UnspentOutput({
+        txid: coin.hash,
+        vout: coin.index,
+        address: coin.address,
+        script: coin.script,
+        amount: coin.value * 0.00000001 // Satoshi amount
+      });
+    });
+    console.info(`Returning ${unspentTransactions.length} coins`);
+    return unspentTransactions;
+  }
+
+  /**
+   * Broadcasts a transaction to the bitcoin network
+   * @param transaction Transaction to broadcast
+   */
+  private async broadcastTransaction (transaction: Transaction): Promise<boolean> {
+    const rawTransaction = transaction.serialize();
+    console.info(`Boradcasting transaction ${transaction.id}`);
+    const request = JSON.stringify({
+      tx: rawTransaction
+    });
+    const fullPath = new URL('/broadcast', this.bitcoinExtensionUri);
+    const responseObject = await this.fetchWithRetry(fullPath.toString(), {
+      body: request,
+      method: 'post'
+    });
+    const responseData = await ReadableStreamUtils.readAll(responseObject.body);
+    if (responseObject.status !== httpStatus.OK) {
+      const error = new Error(`Broadcast failure [${responseObject.status}]: ${responseData}`);
+      console.error(error);
+      throw error;
+    }
+
+    const response = JSON.parse(responseData);
+
+    return response.success;
+  }
+
+  /**
+   * Will process transactions every interval seconds.
+   * @param interval Number of seconds between each query
+   */
+  private periodicPoll (interval: number = this.pollPeriod) {
+    if (this.pollTimeoutId) {
+      clearTimeout(this.pollTimeoutId);
+    }
+    this.processTransactions(this.lastSeenBlock).then((syncedTo) => {
+      this.lastSeenBlock = syncedTo;
+      this.pollTimeoutId = setTimeout(this.periodicPoll.bind(this), 1000 * interval, interval);

```

Comment:
```suggestion
      this.pollTimeout = setTimeout(() => this.periodicPoll(), 1000 * this.pollPeriod,);
```
Consider non "bind" notation. Though you can't just take the suggested line at face value because it has assumption.

Reply:
the `this.periodicPoll()` would lose reference to `this` when called by setTimeout (and theres a trailing comma).